servant-client/test/Servant/Client/TestServer/GHCJS.hs

```diff
@@ -0,0 +1,34 @@
+
+module Servant.Client.TestServer.GHCJS where
+
+import           Control.Exception
+import           Network.Wai
+import           Safe
+import           System.Exit
+import           System.IO
+import           System.Process
+
+import           Servant.Common.BaseUrl
+
+buildTestServer :: IO ()
+buildTestServer = do
+  process <- spawnProcess "./test/ghcjs/build-test-server.sh" []
+  ExitSuccess <- waitForProcess process
+  return ()
+
+withTestServer :: Application -> String -> (BaseUrl -> IO a) -> IO a
```

Comment:
I hope I made this much clearer now by refactoring the code and adding this comment: https://github.com/haskell-servant/servant/commit/8f7dedd90cc92a90688ffb41c851f65cc9b94e1d